tests/system/src/test/java/test/robot/javafx/embed/swing/LinuxScreencastHangCrashTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.embed.swing;
+
+import java.awt.Robot;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javafx.application.Platform;
+import javafx.scene.paint.Color;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.Timeout;
+import test.util.Util;
+
+@TestMethodOrder(OrderAnnotation.class)
+public class LinuxScreencastHangCrashTest {
+
+    private static Robot robot;
+    private static javafx.scene.robot.Robot jfxRobot;
+
+    private static final int DELAY_BEFORE_SESSION_CLOSE = 2000;
+    private static final int DELAY_WAIT_FOR_SESSION_TO_CLOSE = DELAY_BEFORE_SESSION_CLOSE + 250;
+    private static final int DELAY_KEEP_SESSION = DELAY_BEFORE_SESSION_CLOSE - 1000;
+
+    private static volatile boolean isFxStarted = false;
+
+    @BeforeAll
+    public static void init() throws Exception {
+        Assumptions.assumeTrue(!Util.isOnWayland()); // JDK-8335470
```

Comment:
Once the AWT fix is in a promoted build of JDK 24, this can be replaced (here and in the other two test classes that are excluded on Wayland) with:

```
    assumeTrue(Runtime.version().feature() >= 24);
```

I added this as a comment in the bug report for JDK-8335470 as well.